src/neighbourhood.jl

```diff
@@ -0,0 +1,36 @@
+"""
+A neighbourhood function returns a tuple of two elements: 
```

Comment:
a general question: would it make it easier to implement the algorithms if this function returns an expression with a hole at the place that has to be modified, instead of the location of that place?

Reply:
We have discussed it on Mattermost that we can ignore the holes.